tf-modules/data-persistence/variables.tf

```diff
@@ -0,0 +1,71 @@
+# Required
+
+variable "prefix" {
+  description = "prefix to use for naming created resources"
+  type        = string
+}
+
+# Optional
+
+variable "create_service_linked_role" {
+  description = "Whether to create an IAM service linked role for Elasticsearch"
+  type        = bool
+  default     = true
+}
+
+variable "es_trusted_role_arns" {
+  description = "IAM role ARNs that should be trusted for connecting to Elasticsearch"
+  type        = list(string)
+  default     = []
+}
+
+variable "include_elasticsearch" {
+  description = "True/false for whether to deploy Elasticsearch"
+  type        = bool
+  default     = true
+}
+
+variable "elasticsearch_config" {
+  description = "Configuration object for Elasticsearch"
+  type = object({
+    domain_name    = string
+    instance_count = number
+    instance_type  = string
+    version        = string
+    volume_size    = number
+  })
+  default = {
+    domain_name    = "es"
+    instance_count = 1
+    instance_type  = "t2.small.elasticsearch"
+    version        = "5.3"
+    volume_size    = 10
+  }
+}
+
+variable "enable_point_in_time_tables" {
+  description = "DynamoDB table names that should have point in time recovery enabled"
+  type        = list(string)
+  default     = [
+    "CollectionsTable",
+    "ExecutionsTable",
+    "FilesTable",
+    "GranulesTable",
+    "PdrsTable",
+    "ProvidersTable",
+    "RulesTable",
+    "UsersTable"
+  ]
+}
+
+variable "security_groups" {
+  description = "Security Group IDs (for Elasticsearch)"
+  type        = list(string)
+  default     = []
+}
+
+variable "subnet_ids" {
+  description = "Subnet IDs (for Elasticsearch)"
+  type        = list(string)
+  default     = []
```

Comment:
I'm not sure that one is better than another, but I usually set `default = null` and then write my check in the code as `inside_vpc = var.subnet_id != null`

Reply:
If the type is list, then it feels more natural to me that the default should be empty list. In this case we're using the array length to determine another variable, but in other cases I think Terraform is smart enough to handle an empty array for an attribute and do nothing, in which case making that the default seems ideal. So it seems better by convention to stick with an empty array for the default of optional lists.

Does that make sense? I'm not 100% sure it does